### PR TITLE
add tests: encrypt under wrong public key

### PR DESCRIPTION
**Pull request by PeterChen13579**
### Context of Change
Add tests and small checks under confidential MPT

### API Impact

- [ ] Public API: New feature (new methods and/or new fields)
- [ ] Public API: Breaking change (in general, breaking changes should only impact the next api_version)
- [ ] `libxrpl` change (any change that may affect `libxrpl` or dependents of `libxrpl`)
- [ ] Peer protocol change (must be backward compatible or bump the peer protocol version)

### CODE REVIEW

File: src/libxrpl/tx/transactors/token/ConfidentialMPTClawback.cpp [hunk 1]
```diff
@@ -77,6 +81,10 @@ ConfidentialMPTClawback::preclaim(PreclaimContext const& ctx)
     if (!sleHolderMPToken->isFieldPresent(sfIssuerEncryptedBalance))
         return tecNO_PERMISSION;
 
+    // Check if Holder has ElGamal public Key
+    if (!sleHolderMPToken->isFieldPresent(sfHolderEncryptionKey))
+        return tecNO_PERMISSION;
+
```
Comment (shawnxie999) [inline src/libxrpl/tx/transactors/token/ConfidentialMPTClawback.cpp:87]: both returns should have `// LCOV_EXCL_LINE` to exclude them from test coverage since they are defensive checks that are unreachable

File: src/test/app/ConfidentialTransfer_test.cpp [hunk 2]
```diff
@@ -2734,9 +2734,143 @@ class ConfidentialTransfer_test : public beast::unit_test::suite
...
+            .account = bob,
+            .dest = carol,
+            .amt = 0,
+            .proof = getTrivialSendProofHex(3),
+            .senderEncryptedAmt = mptAlice.encryptAmount(bob, 0, bf2),
+            .destEncryptedAmt = mptAlice.encryptAmount(carol, 0, bf2),
+            .issuerEncryptedAmt = mptAlice.encryptAmount(alice, 0, bf2),
+            .amountCommitment = getTrivialCommitment(),
+            .balanceCommitment = getTrivialCommitment(),
+            .err = tecBAD_PROOF,
```
Comment (shawnxie999) [inline src/test/app/ConfidentialTransfer_test.cpp:3017]: do we need to pass in each param for these? or would it work if we just let autofilling do the work? if so, we will probably need update these tests once zero-amount is supported

File: src/test/app/ConfidentialTransfer_test.cpp [hunk 3]
```diff
@@ -6302,6 +6436,89 @@ class ConfidentialTransfer_test : public beast::unit_test::suite
...
+    /* This test ensures that when sending confidential tokens, the encrypted
+     * amounts are securely locked to the correct accounts' official public keys.
+     *
+     * Attack scenario — Encrypting the issuer's copy with the wrong key:
+     * A sender correctly encrypts the hidden transfer amount for themselves
+     * and the receiver. However, they intentionally encrypt the issuer's
+     * copy of the data using the wrong public key (for example, using the
+     * receiver's key instead of the official issuer's key).
+     */
+
```
Comment (shawnxie999) [inline src/test/app/ConfidentialTransfer_test.cpp]: extra line

File: src/test/jtx/impl/mpt.cpp [hunk 4]
```diff
@@ -1309,6 +1309,20 @@ MPTTester::send(MPTConfidentialSend const& arg)
...
+        // fails isValidCompressedECPoint in preflight. For spending=0,
+        // PC = 0*H + r*G = r*G — compute it directly.
+        secp256k1_pubkey pt;
+        auto const sCtx = secp256k1Context();
+        if (secp256k1_ec_pubkey_create(sCtx, &pt, balanceBlindingFactor.data()) != 1)
+            Throw<std::runtime_error>("Failed to create balance commitment for zero balance");
+        balanceCommitment = Buffer(ecPedersenCommitmentLength);
+        size_t len = ecPedersenCommitmentLength;
+        secp256k1_ec_pubkey_serialize(
+            sCtx, balanceCommitment.data(), &len, &pt, SECP256K1_EC_COMPRESSED);
```
Comment (shawnxie999) [inline src/test/jtx/impl/mpt.cpp]: Is this code used right now at the moment? I don't think this code is logically correct

Comment (PeterChen13579) [reply src/test/jtx/impl/mpt.cpp]: Ah, this code can be deleted. I was trying to find a way to bypass the current failure of getPedersenCommitment(0,) before we discussed crypto lib will fix it. 
Thanks for the catch!

File: src/test/app/ConfidentialTransfer_test.cpp (hunk 2 above)
Comment (PeterChen13579) [reply src/test/app/ConfidentialTransfer_test.cpp:3017]: Let's keep this as is for now (since the code is commented out); I will revisit in the future when confidentialSend with Amt 0 is supported.
